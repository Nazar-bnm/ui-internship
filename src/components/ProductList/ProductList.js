/* eslint-disable react/no-did-update-set-state */
import React from 'react';

import ProductListNavigationContainer from './ProductListNavigation';
import ItemsListContainer from './ItemsList';
import Pagination from '../Pagination';

import './ProductList.scss';

export const CN = 'product-list';

class ProductList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      dropdownSortingSelectedID: '',
      dropdownItemsNumberSelectedID: '',
      currentPage: 1
    };

    this.setCurrentPage = this.setCurrentPage.bind(this);
  }

  componentDidUpdate({ filterProps: prevFilterProps }) {
    const { filterProps } = this.props;

    const isChangedFilters = prevFilterProps.some(
      (prevFilterProp, index) => filterProps[index] !== prevFilterProp
    );

    isChangedFilters
      && this.setState({
        currentPage: 1
      });
  }

  setCurrentPage(currentPage) {
    this.setState({
      currentPage
    });
  }

  changeSortingSelectedID = (selectedID) => {
    this.setState({
      dropdownSortingSelectedID: selectedID
    });
  };

  changeItemsNumberSelectedID = (selectedID) => {
    const { dropdownItemsNumberSelectedID } = this.state;

    selectedID !== dropdownItemsNumberSelectedID
      && this.setState({
        dropdownItemsNumberSelectedID: selectedID,
        currentPage: 1
      });
  };

  render() {
    const {
      dropdownSortingSelectedID,
      dropdownItemsNumberSelectedID,
      currentPage
    } = this.state;

    const { gender, length, chosenItemsOnPage } = this.props;
    const numberOfPages = Math.ceil(length / chosenItemsOnPage);
    const visibleNumbers = numberOfPages > 5 ? 5 : 3;

    return (
      <div className={`${CN}`}>
        <div className={`${CN}__filter-wrapper`}>
          <Pagination
            currentPage={currentPage}
            numberOfPages={numberOfPages}
            setCurrentPage={this.setCurrentPage}
            visibleNumbers={visibleNumbers}
          />
          <ProductListNavigationContainer
            dropdownSortingSelectedID={dropdownSortingSelectedID}
            dropdownItemsNumberSelectedID={dropdownItemsNumberSelectedID}
            changeSortingSelectedID={this.changeSortingSelectedID}
            changeItemsNumberSelectedID={this.changeItemsNumberSelectedID}
          />
        </div>
        <div className={`${CN}__items-list-container`}>
          <ItemsListContainer gender={gender} currentPage={currentPage - 1} />
        </div>
        <div className={`${CN}__filter-wrapper`}>
          <Pagination
            currentPage={currentPage}
            numberOfPages={numberOfPages}
            setCurrentPage={this.setCurrentPage}
            visibleNumbers={visibleNumbers}
          />
          <ProductListNavigationContainer
            dropdownItemsNumberSelectedID={dropdownItemsNumberSelectedID}
            dropdownSortingSelectedID={dropdownSortingSelectedID}
            changeSortingSelectedID={this.changeSortingSelectedID}
            changeItemsNumberSelectedID={this.changeItemsNumberSelectedID}
          />
        </div>
      </div>
    );
  }
}

export default ProductList;
