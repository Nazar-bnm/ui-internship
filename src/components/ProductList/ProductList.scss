@import '../../scss/global_styles';

.product-list {
  @include flex($direction: column, $justify: space-between);
  padding: 20px 2vw;

  &__filter-wrapper {
    @include flex($align: center, $justify: space-between);
    width: 100%;
    margin: 10px 0;
  }

  &__list-items-container {
    @include flex();
    width: 100%;
  }

  &__dropdowns-wrapper {
    @include flex();
  }

  &__dropdowns-labels {
    color: $dove-gray;
    font-size: 0.7rem;
    font-family: font(primary-bold);
    font-weight: bold;
    text-transform: uppercase;
  }

  &__dropdowns-labels-wrapper {
    @include flex($align: center, $wrap: nowrap);
    margin: 5px 10px;
  }

  .dropdown {
    @include flex($align: baseline);
    margin: 0 10px;
    border: 1px solid $silver-chalice;
    color: $dove-gray;
    font-size: 0.8rem;

    &__items__element, .dropdown[aria-expanded=true] .dropdown__items__element {
      background-color: $white;
      z-index: 1;
    }
  }

  &__arrow-btn {
    background-color: transparent;
    border: none;

    & i {
      margin: 0 5px;
      color: $silver-chalice;
    }
  }
}