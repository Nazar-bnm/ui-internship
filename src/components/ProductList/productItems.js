/* eslint-disable max-len */
export const itemsArray = [
  {
    id: 1,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'bA',
    collection: 'spring collection 2016',
    category: 'women',
    price: 1,
  },
  {
    id: 2,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'aB',
    collection: 'spring collection 2016',
    category: 'women',
    price: 2,
  },
  {
    id: 3,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'aC',
    collection: 'spring collection 2016',
    category: 'women',
    price: 3,
  },
  {
    id: 4,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'D',
    collection: 'spring collection 2016',
    category: 'women',
    price: 4,
  },
  {
    id: 5,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'E',
    collection: 'spring collection 2016',
    category: 'women',
    price: 5,
  },
  {
    id: 6,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'H',
    collection: 'spring collection 2016',
    category: 'women',
    price: 6,
  },
  {
    id: 7,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'I',
    collection: 'spring collection 2016',
    category: 'women',
    price: 7,
  },
  {
    id: 8,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'J',
    collection: 'spring collection 2016',
    category: 'women',
    price: 8,
  },
  {
    id: 9,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'aaK',
    collection: 'spring collection 2016',
    category: 'women',
    price: 9,
  },
  {
    id: 10,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'AL',
    collection: 'spring collection 2016',
    category: 'women',
    price: 10,
  },
  {
    id: 11,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'M',
    collection: 'spring collection 2016',
    category: 'women',
    price: 11,
  },
  {
    id: 12,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'N',
    collection: 'spring collection 2016',
    category: 'women',
    price: 12,
  },
  {
    id: 13,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'O',
    collection: 'spring collection 2016',
    category: 'women',
    price: 13,
  },
  {
    id: 14,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'P',
    collection: 'spring collection 2016',
    category: 'women',
    price: 14,
  },
  {
    id: 15,
    photo: 'https://www.next.ua/nxtcms/resource/image/1889302/portrait_ratio1x1/525/525/2fccc457347d46d0c506ee9ea0396ae8/of/emma-willis.jpg',
    title: 'R',
    collection: 'spring collection 2016',
    category: 'women',
    price: 15,
  },
];

