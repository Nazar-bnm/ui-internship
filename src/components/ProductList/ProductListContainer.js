import { connect } from 'react-redux';

import ProductList from './ProductList';

const mapStateToProps = (state) => ({
  length: state.reducerProductList.length,
  chosenItemsOnPage: Number(state.reducerProductList.chosenItemsOnPage),
  price: state.reducerFilter.price,
  filterProps: [
    state.reducerFilter.bottoms.length,
    state.reducerFilter.tops.length,
    state.reducerFilter.sizes.length,
    state.reducerFilter.colors.length,
    state.reducerFilter.brands.length
  ]
});

export default connect(mapStateToProps)(ProductList);
