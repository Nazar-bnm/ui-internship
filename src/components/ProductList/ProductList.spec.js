import React from 'react';
import { shallow } from 'enzyme';
import ProductList from './ProductList';

describe('<ProductList />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <ProductList />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});