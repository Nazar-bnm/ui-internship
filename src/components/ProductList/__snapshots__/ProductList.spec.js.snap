// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList /> should match snapshot 1`] = `
<div
  className="product-list content"
>
  <div
    className="product-list__filter-wrapper"
  >
    <React.Memo />
  </div>
  <div
    className="product-list__items-list-container"
  >
    <React.Memo />
  </div>
  <div
    className="product-list__filter-wrapper"
  >
    <React.Memo />
  </div>
</div>
`;
